src/melee/it/types.h

```diff
@@ -0,0 +1,854 @@
+                             // update position if toggled ON
+    UnkFlagStruct x25_flags;
+    UnkFlagStruct x26_flags;
+    UnkFlagStruct x27_flags;
+    Vec3 x28_hurt1_pos;
+    Vec3 x34_hurt2_pos;
+    s32 x40_bone_index;
+
+} itHurt;
+
+typedef struct itHitVictim {
+    void* x0_victim;   // m-ex header says this is a user_data pointer but
+                       // proceeds to call it a GObj in the comment below it
+    s32 x4_timedRehit; // number of frames needed to pass before this entity can
+                       // be hit again; 0 = can't rehit
+} itHitVictim;
+
+typedef struct itHit {
+    BOOL x0_toggle; // Toggles hitbox on/off.
+    s32 x4_unk;
+    s32 x8_damage;        // Projected damage
+    f32 xC_damage_staled; // Staled damage, actually applied
+    Vec3 x10_offset;
+    f32 x1C_hitbox_size;
+    s32 x20_angle;
+    s32 x24_knockback_growth;
+    s32 x28_wdsk; // Weight-Dependent Set Knockback
+    s32 x2C_base_knockback;
+    s32 x30_element;       // Normal, fire, electric, etc.
+    s32 x34_shield_damage; // If hitbox damage + shield damage is less than 0
+                           // (negative), this will effectively restore shield
+                           // health
+    s32 x38_SFX_severity;  // 0x38. hurtbox interaction. 0 = none, 1 = grounded,
+                           // 2 = aerial, 3 = both // What
+    s32 x3C_SFX_kind;
+    UnkFlagStruct x40_flags; // 0x20 -> check against aerial fighters; 0x10 ->
+                             // check against grounded fighters
+    UnkFlagStruct x41_flags; // 0x8/0x4/0x2 = timed rehit on
+                             // item/fighter/shield; 0x1 = can reflect
+    UnkFlagStruct x42_flags; // 0x80 = can absorb; 0x20 = hit only fighters
+                             // facing the item; 0x10 = can shield; 0x8 = ignore
+                             // reflect/absorb bubbles(?); 0x4 = ignore
+                             // hurtboxes; 0x2 = ignore ungrabbable hurtboxes
+    UnkFlagStruct
+        x43_flags; // 0x80 = interact with items only; 0x20 = interact with all?
+    s32 x44;
+    HSD_JObj* x48_jobj;
+    Vec3 pos;
+    Vec3 x58_posPrev;
+    Vec3 x64_posColl;      // 0x64   position of hurt collision
+    f32 x70_coll_distance; // 0x70   Distance From Collding Hurtbox (Used for
+                           // phantom hit collision calculation)
+    itHitVictim x74_tipLog[12];
+    itHitVictim xD4_damageLog[12];
+    s32 x134;
+    s32 x138;
+} itHit;
+
+typedef enum ItSfx {
+    ItSfxNone = -1,
+} ItSfx;
+
+typedef struct Item {
+    void* x0;
+    HSD_GObj* x4_GObj;
+    s32 x8;
+    s32 xC_spawn_kind;
+    s32 x10_item_kind;
+    enum_t hold_kind;
+    s32 x18;
+    s32 x1C;
+    u8 x20_team_id;
+    u8 x21;
+    u8 x22;
+    u8 x23;
+    s32 x24_item_state_index;
+    s32 x28_item_anim_index;
+
+    /// @at{2C} @sz{4}
+    f32 facing_dir;
+
+    f32 init_facing_dir;
+    f32 x34_spin_speed;
+    f32 x38_scale;
+    f32 x3C;
+    Vec3 x40_vel;
+    Vec3 pos;
```

Comment:
Oh yeah, and if we're dropping the hex offsets from the name, let's keep the offset in a comment until decomp is done.

Reply:
Oh, missed that one.

Reply:
Also needed for `hold_kind`, `owner`, and `init_facing_dir`.